smdebug/core/hook.py

```diff
@@ -652,8 +654,17 @@ def _write_scalars(self):
                         scalar_name, scalar_val, self.step, timestamp=timestamp
                     )
             if write_event:
-                if self.writer is None:
+                writer_not_initialized = False
+                if isinstance(self, BaseHook):
+                    writers = self._get_writers(None, None)
+                    if not any(writers):
+                        writer_not_initialized = True
+                elif self.writer is None:
+                    writer_not_initialized = True
+
+                if writer_not_initialized:
```

Comment:
instead of these if checks, would it be possible to modify core/hook.py:_initialize_writers to   _initialize_writers(self, only_initialize_if_missing=False) ?

that way you can just call    self._initialize_writers(self, only_initialize_if_missing=True) here?

Reply:
You're right. This is much cleaner.